src/components/TodoItem.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from '../TodosContext';
+import { deleteTodo } from '../api/todos';
+
+interface Props {
+  todo: Todo
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { setTodos, setErrorMessage } = useContext(TodosContext);
+  const [loading] = useState(false);
+
+  const { id, title, completed } = todo;
+
+  const handleCatch = () => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    deleteTodo(todoId)
+      .catch(() => setErrorMessage('Unable to delete a todo'))
+      .finally(() => handleCatch());
+
+    setTodos((prev) => prev.filter((t) => t.id !== todoId));
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames(
+        'todo',
+        // eslint-disable-next-line quote-props
+        { 'completed': completed },
```

Comment:
better write it like in the example below, and remove eslint-disable, it means the same as what you write

```suggestion
      className={classNames(
        'todo',
        { completed },
```